pkg/store/promproxy.go

```diff
@@ -1,36 +1,138 @@
 package store
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
+	"fmt"
 	"net/http"
+	"net/url"
+	"path"
+	"sort"
+	"strconv"
+	"time"
 
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
 	"github.com/improbable-eng/promlts/pkg/store/storepb"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/tsdb/chunks"
 )
 
 // PrometheusProxy implements the store node API on top of the Prometheus
 // HTTP v1 API.
 type PrometheusProxy struct {
-	addr   string
+	base   *url.URL
 	client *http.Client
 }
 
 var _ storepb.StoreServer = (*PrometheusProxy)(nil)
 
 // NewPrometheusProxy returns a new PrometheusProxy that uses the given HTTP client
 // to talk to Prometheus.
-func NewPrometheusProxy(client *http.Client, addr string) *PrometheusProxy {
-	return &PrometheusProxy{addr: addr, client: client}
+func NewPrometheusProxy(client *http.Client, baseURL string) (*PrometheusProxy, error) {
+	if client == nil {
+		client = http.DefaultClient
+	}
+	u, err := url.Parse(baseURL)
+	if err != nil {
+		return nil, errors.Wrap(err, "parse base URL")
+	}
+	return &PrometheusProxy{base: u, client: client}, nil
 }
 
 // Series returns all series for a requested time range and label matcher. The returned data may
 // exceed the requested time bounds.
 func (p *PrometheusProxy) Series(ctx context.Context, r *storepb.SeriesRequest) (
 	*storepb.SeriesResponse, error,
 ) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	// Prometheus's range query API is not suitable to give us all datapoints. We use the
+	// instant API and do a range selection in PromQL to cover the queried time range.
```

Comment:
I think that whole comment block should go to `Series` method description. It is not related to `args := url.Values{}`